### PR TITLE
[JENKINS-32801] implements SimpleBuildStep

### PR DESCRIPTION
**Pull request by kuisathaverat**
https://issues.jenkins-ci.org/browse/JENKINS-32801

It adds compatibility with Pipeline plugin, now you can run a step of Zephyr fro jita test management plugin.

```
node {
  git url: 'file:////Users/inifc/Documents/workspace/statusCheck'
  def mvnHome = tool 'maven3'
  sh "${mvnHome}/bin/mvn clean package"
  step([$class: 'JUnitResultArchiver', testResults: '**/target/surefire-reports/TEST-*.xml'])
  step([$class: 'ZfjReporter',serverAddress: 'http://127.0.0.1:9090',projectKey:'PRJ',versionKey:'1',cycleKey:'CreateNewCycle',cycleDuration:'30 days',cyclePrefix:''])
}
```

@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: src/main/java/com/thed/zephyr/jenkins/reporter/ZfjReporter.java
Comment (amuniz) [inline src/main/java/com/thed/zephyr/jenkins/reporter/ZfjReporter.java]: Why did you add this constants? It's weird to use constants for log messages, if the same message is used repeatedly you can just add a log method (i.e. `logSurefireError()`, `logJobConfigurationError()`, etc)

**Comment by amuniz** [comment]
LGTM. I just miss a pipeline specific test (there are examples in - almost - all plugins pipeline-compatible, for example [here](https://github.com/jenkinsci/findbugs-plugin/blob/4b429d04284200a4740b02e530289050489eb2fb/plugin/src/test/java/hudson/plugins/findbugs/workflow/WorkflowCompatibilityTest.java#L30))

**Comment by kuisathaverat** [comment]
@mohan-zephyr Could you evaluate to merge this pull request? It makes this plugin compatible with Pipeline Plugin and the new paradigm of Jenkins Pipeline-as-Code

**Comment by gregamb** [comment]
@mohan-zephyr I second @kuisathaverat request :+1:

**Comment by ghost** [comment]
Hi there,
are there any plans on merging this pr somewhere son? Would be really great to have this feature... or is there a workaround ?
Greetings 
Felix

**Comment by zackpollard** [comment]
+1 for getting this implemented @kuisathaverat there is now a merge conflict with the PR though

**Review by oleg-nenashev — APPROVED** [review]
Needs a merge conflict resolution, but 🐝 in general.
@reviewbybees done

**Comment by zackpollard** [comment]
@oleg-nenashev I have resolved the merge conflicts in a PR I made to @kuisathaverat branch. So if he accepts that then they should fall through to this PR too :)

**Comment by kuisathaverat** [comment]
@zackpollard I saw too many changes on your PR only to resolve a conflict on one file I will review the conflict and try to resolve it without too many changes

**Comment by zackpollard** [comment]
@kuisathaverat That's because it's also PRing all of the changes that your branch doesn't have from the master branch in this repository. Once they get merged into this PR, you will no longer see any of the commits other than the merge

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Seems the merge was not correct 🐛 
`Caused by: org.apache.maven.plugin.CompilationFailureException: Compilation failure
/scratch/jenkins/workspace/plugins/zephyr-for-jira-test-management-plugin/src/main/java/com/thed/zephyr/jenkins/reporter/ZfjReporter.java:[375,26] error: cannot find symbol`

**Comment by zackpollard** [comment]
@oleg-nenashev yea there's an issue with one of the new calls that didn't show up when I initially merged... Just looking into the fix now

**Comment by zackpollard** [comment]
@oleg-nenashev @kuisathaverat These changes seem odd, they are under a "Temp commit" title https://github.com/jenkinsci/zephyr-for-jira-test-management-plugin/commit/42362605c999c3f1590d9f924441a54bc76a9736
The getRootNumber() is breaking the build

**Comment by kuisathaverat** [comment]
it is because it wants an AbstractBuild not a Run class

**Comment by zackpollard** [comment]
@kuisathaverat Yea, however i'm not sure we can pipe that through to there. Looking at the Run class, I think the getNumber() from that should suffice for getting the build number. We shouldn't need the AbstractBuild

**Comment by kuisathaverat** [comment]
@zackpollard me too, but I do not understand why it use getRootBuild for me does not have too much sense 

> public AbstractBuild<?,?> getRootBuild()
> Gets the nearest ancestor AbstractBuild that belongs to the root project of getProject() that dominates/governs/encompasses this build.
> Some projects (such as matrix projects, Maven projects, or promotion processes) form a tree of jobs, and still in some of them, builds of child projects are related/tied to that of the parent project. In such a case, this method returns the governing build.

**Comment by zackpollard** [comment]
@kuisathaverat Yea i'm unsure why you'd want to go up to the root build, I think the exact equivalent for this of run would be `getProject().getNumber()` however maybe we should loop in the person who made those changes originally for context?

**Comment by zackpollard** [comment]
Failure on the build due to findbugs task
[ERROR] Failed to execute goal org.codehaus.mojo:findbugs-maven-plugin:3.0.3:check (findbugs) on project zephyr-for-jira-test-management: failed with 53 bugs and 0 errors

**Comment by kuisathaverat** [comment]
none of those are in ZfjReporter.java that it is the only file that was changed so I am not gonna to touch any other file

**Comment by zackpollard** [comment]
Yep, so how do we proceed from here? 

Sent from MailDroid

-----Original Message-----

**Comment by kuisathaverat** [comment]
we could open another PR for the findbugs issues but I don't know if the maintainer will attend it because this was open for more than a year

**Comment by zackpollard** [comment]
So what now then, considering the bugs found are not with this class, then surely this PR could proceed regardless?

**Comment by kuisathaverat** [comment]
I made other PR to fix some of the findbugs findings https://github.com/jenkinsci/zephyr-for-jira-test-management-plugin/pull/3

**Comment by lbrigman124** [comment]
Outside of the failing tests, The code fails on my system when trying to process JUnit results.
Same plugin on the same system when running within a freestyle job works.
java.lang.NullPointerException
	at com.thed.zephyr.jenkins.reporter.ZfjReporter.perform(ZfjReporter.java:363)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:80)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:67)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1$1.call(SynchronousNonBlockingStepExecution.java:49)
	at hudson.security.ACL.impersonate(ACL.java:221)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution$1.run(SynchronousNonBlockingStepExecution.java:46)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:748)

**Comment by kuisathaverat** [comment]
@lbrigman124 this PR does not pass the find bug checks https://jenkins.ci.cloudbees.com/job/plugins/job/zephyr-for-jira-test-management-plugin/23 , there is another PR to fix these errors https://github.com/jenkinsci/zephyr-for-jira-test-management-plugin/pull/3 but probably both are really outdated after more than a year, the maintainer does not have interest on those PR me neither

**Comment by Olegwitcher** [comment]
Hello @kuisathaverat .
If I correctly understand comments - We are not planning to add this plugin to support list of Jenkins pipeline.  And for now - we don't have any ability trigger Zephyr through Jenkins step?

**Comment by kuisathaverat** [comment]
It is in the maintainer roof, this PR is no longer valid to make it mergeable, we have to make a ton of fix related with findbugs and all the changes in the code since it was proposed.

**Comment by Olegwitcher** [comment]
Thank you for answer , @kuisathaverat

**Comment by sameerb29** [comment]
is there any workaround to add zephyr for jira in Jenkins pipeline ??

**Comment by ChioSoCo** [comment]
> is there any workaround to add zephyr for jira in Jenkins pipeline ??

hey @sameerb29 , did you find a workaround for this?

**Comment by nishant041** [comment]
Any solution for this?

**Comment by sahil-zymr** [comment]
Any updates on this issue?

**Pull request closed without merging**
